storage-access.bs

```diff
@@ -45,33 +47,69 @@ spec:html; type:dfn; text:current entry; url:https://html.spec.whatwg.org/multip
+User Agents sometimes prevent content inside certain <{iframe}>s from accessing data stored in client-side storage mechanisms like cookies. This can break embedded content which relies on having access to client-side storage.
 
-ISSUE: [cross origin or cross site?](https://github.com/privacycg/storage-access/pull/24#discussion_r408771982)
-
-ISSUE: give other examples beyond authenticated embeds
+The Storage Access API enables content inside <{iframe}>s to request and be granted access to their client-side storage, so that embedded content which relies on having access to client-side storage can work in such User Agents. [[STORAGE-ACCESS-INTRO]]
 
 </section>
 
 <h2 id="infra">Infrastructure</h2>
 
-This specification defines several additions to the HTML standard, and depends on the Infra standard. [[!INFRA]] [[!HTML]]
+This specification depends on the Infra standard. [[!INFRA]]
 
 <h2 id="the-storage-access-api">The Storage Access API</h2>
 
+A {{Document}} is said to be <dfn>first party</dfn> when the <a for="environment settings object">origin</a> and [=top-level origin=] of its [=relevant settings object=] are [=same site=] with one another. Otherwise, it is said to be <dfn>third party</dfn>.
+
+The <dfn>first party data</dfn> of a {{Document}} is the client-side storage it would normally have access to if it were [=first party=].
+
+This specification defines a method to query whether or not a {{Document}} currently has access to its [=first party data=] ({{Document/hasStorageAccess()}}), and a method that can be used to request access to a {{Document}}'s [=first party data=] ({{Document/requestStorageAccess()}}).
+
+<div class=example>
+
+Alex visits `https://social.example/`. The page sets a cookie. This cookie has been set in a first party context.
+
+Later on, Alex visits `https://video.example/`, which has an <{iframe}> on it which loads `https://social.example/heart-button-widget`. In this case, the `social.example` {{Document}} |doc| is [=third party=], and the cookie set previously might or might not be visible from |doc|`.`{{Document/cookie}}, depending on User Agent storage access policies.
+
+Script in the <{iframe}> can call |doc|`.`{{Document/hasStorageAccess()}} to determine if it has access to the cookie. If it does not have access, it can request access by calling |doc|`.`{{Document/requestStorageAccess()}}.
+
+</div>
+
+<h3 id="ua-state">User Agent state related to storage access</h3>
+
+Each User Agent maintains a <dfn>storage access map</dfn>, an [=ordered map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
+
+ISSUE: What's the lifetime of the [=storage access map=] and of its entries?
+
+A <dfn>partitioned storage key</dfn> is a [=pair=] consisting of a top-level domain and a domain, both [=registrable domains=].
```

Comment:
You want HTML's site concept here, I think. Otherwise you would not have storage if you're visiting an IP address directly, for instance. (And it would also not isolate HTTP from HTTPS.)

Reply:
I've adopted this concept in fa8f82e. Let me know if that works.